interpreter/cling/lib/Interpreter/TransactionPool.h

```diff
@@ -73,15 +73,14 @@ namespace cling {
       if (T->getParent())
         T->getParent()->removeNestedTransaction(T);
 
-      T->~Transaction();
-
       // don't overflow the pool
       if (reuse && (m_Transactions.size() < kPoolSize)) {
         T->m_State = Transaction::kNumStates;
+        T->~Transaction();
```

Comment:
Since my comment was going to be: "add a comment in ~Transaction to warn to not modify m_state", I think it is actually better to migrate line 78 into the destructor and be done.

Reply:
I will not touch this here, a follow up OR will change all of this... I could move it into the destructor, but then tooling could rightfully complain that this is a useless statement.